### PR TITLE
Use spotbugs-enabled parent pom, fix spotbugs errors.

### PR DESCRIPTION
**Pull request by brianloss**
Fix spotbugs warnings (and a couple other intellij-identified issues).

### CODE REVIEW

File: api/src/main/java/datawave/webservice/request/objects/ReferencedValue.java [hunk 1]
```diff
@@ -82,7 +83,7 @@ public void setValue(String value) {
         if (isValidXML(value)) {
             this.value = value;
         } else {
-            this.value = new String(Base64.encodeBase64(value.getBytes(Charset.forName("UTF-8"))));
+            this.value = new String(Base64.encodeBase64(value.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
```
Comment (ctubbsii) [inline api/src/main/java/datawave/webservice/request/objects/ReferencedValue.java]: FWIW, I find it very useful to statically import the `UTF_8` field, so these things can be shorter. Guava also had a `UTF_8` instance, and it was really nice to transition from Guava to the Java built-in by only changing the static import. Also, the line is shorter when the import is static. I know sometimes they are discouraged, but this is one of those rare cases, I think, where it's a good idea to use static imports.

File: api/src/main/java/datawave/webservice/response/objects/DefaultKey.java [hunk 2]
```diff
@@ -43,34 +43,34 @@ public DefaultKey() {
     
     @Override
     public String getRow() {
-        if (this.row.getType().equals(TypedValue.XSD_STRING) && this.row.isBase64Encoded() == true) {
-            return new String(Base64.decodeBase64(this.row.getValue().toString().getBytes(Charset.forName("UTF-8"))));
+        if (this.row.getType().equals(TypedValue.XSD_STRING) && this.row.isBase64Encoded()) {
+            return new String(Base64.decodeBase64(this.row.getValue().toString().getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
```
Comment (ctubbsii) [inline api/src/main/java/datawave/webservice/response/objects/DefaultKey.java]: The modernizer-maven-plugin is also good at catching some of these string-based charsets, and suggesting the concrete type/instance, as well as identifying lots of other code-modernizations that can be done with newer versions of Java. It's definitely worth checking out, if you want another tool beyond spotbugs to help clean up internal tech-debt.

**Review by ctubbsii — DISMISSED** [review]
I recommend using static imports for `UTF_8`, because it makes the code significantly cleaner, but otherwise, this PR looks good to me.

File: api/src/main/java/datawave/webservice/request/objects/ReferencedValue.java (hunk 1 above)
Comment (brianloss) [reply api/src/main/java/datawave/webservice/request/objects/ReferencedValue.java]: Thanks for the feedback! Added.

File: api/src/main/java/datawave/webservice/response/objects/DefaultKey.java (hunk 2 above)
Comment (brianloss) [reply api/src/main/java/datawave/webservice/response/objects/DefaultKey.java]: Will take a look. Thanks!

**Review by keith-ratcliffe — APPROVED** [review]

**Pull request merged**